s2rest_netty/src/main/scala/Server.scala

```diff
@@ -0,0 +1,213 @@
+          simpleResponse(ctx, Ok, byteBufOpt = Option(buf), channelFutureListenerOpt = Option(Close), headers = defaultHeaders)
+        case e: Exception =>
+          logger.error(s"${body}, ${e.getMessage}", e)
+          val buf: ByteBuf = Unpooled.copiedBuffer(PostProcess.emptyResults.toString, CharsetUtil.UTF_8)
+          simpleResponse(ctx, InternalServerError, byteBufOpt = Option(buf), channelFutureListenerOpt = Option(Close), headers = defaultHeaders)
+      }
+    }
+  }
+
+  override def channelRead0(ctx: ChannelHandlerContext, req: FullHttpRequest): Unit = {
+    val uri = req.getUri
+
+    req.getMethod match {
+      case HttpMethod.GET =>
+        uri match {
+          case "/health_check.html" =>
+            if (NettyServer.isHealthy) {
+              val healthCheckMsg = Unpooled.copiedBuffer(NettyServer.deployInfo, CharsetUtil.UTF_8)
+              simpleResponse(ctx, Ok, byteBufOpt = Option(healthCheckMsg), channelFutureListenerOpt = Option(Close))
+            } else {
+              simpleResponse(ctx, NotFound, channelFutureListenerOpt = Option(Close))
+            }
+
+          case s if s.startsWith("/graphs/getEdge/") =>
+            // src, tgt, label, dir
+            val Array(srcId, tgtId, labelName, direction) = s.split("/").takeRight(4)
+            val params = Json.arr(Json.obj("label" -> labelName, "direction" -> direction, "from" -> srcId, "to" -> tgtId))
+            val startedAt = System.currentTimeMillis()
+            val future = s2rest.checkEdges(params)
+            toResponse(ctx, req, params, future.map(_ -> ""), startedAt)
+          case _ => badRoute(ctx)
+        }
+
+      case HttpMethod.PUT =>
+        if (uri.startsWith("/health_check/")) {
+          val newHealthCheck = uri.split("/").last.toBoolean
+          NettyServer.isHealthy = newHealthCheck
+          val newHealthCheckMsg = Unpooled.copiedBuffer(NettyServer.isHealthy.toString, CharsetUtil.UTF_8)
+          simpleResponse(ctx, Ok, byteBufOpt = Option(newHealthCheckMsg), channelFutureListenerOpt = Option(Close))
+        } else badRoute(ctx)
+
+      case HttpMethod.POST =>
+        val jsonString = req.content.toString(CharsetUtil.UTF_8)
+        val jsQuery = Json.parse(jsonString)
+
+        //TODO: result_size
+        val startedAt = System.currentTimeMillis()
+
+        val future =
+          if (uri.startsWith("/graphs/experiment")) {
+            val Array(accessToken, experimentName, uuid) = uri.split("/").takeRight(3)
+            s2rest.experiment(jsQuery, accessToken, experimentName, uuid)
+          } else {
+            s2rest.uriMatch(uri, jsQuery).map(_ -> "")
+          }
+
+        toResponse(ctx, req, jsQuery, future, startedAt)
+
+      case _ =>
+        simpleResponse(ctx, BadRequest, byteBufOpt = None, channelFutureListenerOpt = Option(Close))
+    }
+  }
+
+  override def exceptionCaught(ctx: ChannelHandlerContext, cause: Throwable) {
```

Comment:
RestHandler don't throw exception.
It is returned only Future with status(success, failed)